Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/Constraints/ConstraintManager.cs

```diff
@@ -21,33 +21,48 @@ public ConstraintManager(GameObject gameObject)
             constraints = gameObject.GetComponents<TransformConstraint>().ToList();
         }
 
-        public void ApplyScaleConstraints(ref MixedRealityTransform transform)
+        public void ApplyScaleConstraints(ref MixedRealityTransform transform, bool isOneHanded, bool isNear)
         {
+            ManipulationHandFlags handMode = isOneHanded ? ManipulationHandFlags.OneHanded : ManipulationHandFlags.TwoHanded;
+            ManipulationProximityFlags proximityMode = isNear ? ManipulationProximityFlags.Near : ManipulationProximityFlags.Far;
+            
             foreach (var constraint in constraints)
             {
-                if (constraint.ConstraintType == Utilities.TransformFlags.Scale)
+                if (constraint.ConstraintType == Utilities.TransformFlags.Scale &&
+                    constraint.HandType.HasFlag(handMode) &&
+                    constraint.ProximityType.HasFlag(proximityMode))
                 {
                     constraint.ApplyConstraint(ref transform);
                 }
             }
         }
 
-        public void ApplyRotationConstraints(ref MixedRealityTransform transform)
+        public void ApplyRotationConstraints(ref MixedRealityTransform transform, bool isOneHanded, bool isNear)
         {
+            ManipulationHandFlags handMode = isOneHanded ? ManipulationHandFlags.OneHanded : ManipulationHandFlags.TwoHanded;
+            ManipulationProximityFlags proximityMode = isNear ? ManipulationProximityFlags.Near : ManipulationProximityFlags.Far;
+            
             foreach (var constraint in constraints)
             {
-                if (constraint.ConstraintType == Utilities.TransformFlags.Rotate)
+                if (constraint.ConstraintType == Utilities.TransformFlags.Rotate &&
+                    constraint.HandType.HasFlag(handMode) &&
+                    constraint.ProximityType.HasFlag(proximityMode))
                 {
                     constraint.ApplyConstraint(ref transform);
                 }
             }
         }
 
-        public void ApplyTranslationConstraints(ref MixedRealityTransform transform)
+        public void ApplyTranslationConstraints(ref MixedRealityTransform transform, bool isOneHanded, bool isNear)
```

Comment:
QQ - should the isOneHanded and isNear values be consistent across all invocations for ApplyScaleConstraints and ApplyRotationConstraints and ApplyTranslationConstraints? i.e. are there cases where translation constraints are going to be two handed, but rotation constraints are one handed?

Reply:
Nope, these represent the type of manipulation, so they should all be called with the same values for `isOneHanded` and `isNear`.

In fact, `transform` should be a reference to the same `MixedRealityTransform` for all three calls.